### PR TITLE
new userId module - neustar's fabrick

### PR DESCRIPTION
**Pull request by andersonbd1**
I have verified that all added or changed code includes tests with greater than 80% code coverage.

## Type of change
- [X ] Other - New userId Module - neustar fabrick

## Description of change
submitting userId module for neustar's fabrick - https://www.home.neustar/fabrick

- test parameters for validating userModule
```
pbjs.setConfig({
  userSync: {
    userIds: [
      {
        name: "fabrickId",
        params: {
          apiKey: '789',
          // you can test with this:
          // url: "http://localhost:9999/test/mocks/fabrickId.json?",
          // otherwise it defaults to our actual endpoint
          e: 'abc'
          refererInfo: {
            referer: 'r'.repeat(300),
            stack: ['s-0'],
            canonicalUrl: 'cu-0'
          }
        },
        storage: {
          type: "cookie",
          name: "pbjs-fabrickId",
          expires: 7
        }
      }
    ]}
});
```

- contact email of the adapter’s maintainer - ben.anderson@team.neustar
- [X ] official adapter submission - neustar has recently become an official prebid member.

### CODE REVIEW

**Comment by andersonbd1** [comment]
I tried clicking the "Details" link in the "ci/circleci: build - Your tests failed on CircleCI", which takes me to this link:
https://app.circleci.com/pipelines/github/prebid/Prebid.js/jobs/10112
I am then prompted to login, which I do with my github creds and then it asks me to select "the organization where you'd like to work today" and I'm presented with 2 options - 1) andersonbd1 (my github namespace) and 2) neustar (our company's namespace).
It doesn't matter which I choose, I think get presented with an error screen which just says "Server Error...".  I am wondering if I need to be a member of the prebid organization in circleci to see my error?

**Comment by bretg** [comment]
@andersonbd1 - you shouldn't need to be a member of Prebid to see circle ci, but if you do, that's not going to happen without Neustar becoming a member company. Anyhow, you can get the same results simply by running `gulp test` from the command line.

**Comment by andersonbd1** [comment]
Thanks, Bret.  

> @andersonbd1 - you shouldn't need to be a member of Prebid to see circle ci, 
Then any idea why I'm seeing that error?

> @andersonbd1 -but if you do, that's not going to happen without Neustar becoming a member company. 
We have become members.  (early September, I believe)

> @andersonbd1 Anyhow, you can get the same results simply by running `gulp test` from the command line.  
Ugh - sorry.  I had run `gulp test-coverage`, which I erroneously assumed also ran `gulp test`.  I just ran `gulp test` and see the issues (as you sent).  OK - maybe I'm not as blind as I thought.  I'll get it fixed and resubmitted.  Thanks, Bret.

**Comment by bretg** [comment]
> Then any idea why I'm seeing that error?

Nope. But since you're with a member company, I added you directly to the repo. Try now?

**Comment by andersonbd1** [comment]
Thanks, @bretg .  I resubmitted and can now see my next build error in circleci, which is super helpful because I didn't see that locally (it's an ie issue).

**Comment by bretg** [comment]
Learn something new every day. It's a bummer that people have to have Prebid.org read-level permissions to see the CircleCI output, but I don't see any obvious options in CircleCI here.  C'est la vie. People will need to rely on `gulp test`.

File: test/spec/modules/fabrickIdSystem_spec.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    });
+    let submoduleCallback = fabrickIdSubmodule.getId(configParams).callback;
+    let callBackSpy = sinon.spy();
+    submoduleCallback(callBackSpy);
+    let request = server.requests[0];
+    expect(request.url).to.match(/r=r-0&r=s-0&r=cu-0&r=http/);
+    request.respond(
+      200,
+      responseHeader,
+      // TODO - actually check the value
```
Comment (smenzer) [inline test/spec/modules/fabrickIdSystem_spec.js:92]: do you plan to action this TODO before we merge this PR?

**Review by smenzer — COMMENTED** [review]
Overall LGTM, just one question before we merge

**Comment by smenzer** [comment]
can you also please link to the Docs PR?

File: test/spec/modules/fabrickIdSystem_spec.js (hunk 1 above)
Comment (andersonbd1) [reply test/spec/modules/fabrickIdSystem_spec.js:92]: shoot - meant to take that out.  No, I think it's OK as is.  I can delete and resubmit if you want.

Comment (smenzer) [reply test/spec/modules/fabrickIdSystem_spec.js:92]: if you remove it in your branch and just push and update, the PR will update automatically...no need to resubmit a PR.  i'm fine if it's left in, just wanted to call it out.

**Comment by smenzer** [comment]
@andersonbd1 once you submit a Docs PR I can merge this, but can't until you do so.

File: modules/fabrickIdSystem.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      return { 'fabrickId': value.fabrickId };
+    } else {
+      return undefined;
+    }
+  },
+
+  /**
+   * performs action to obtain id and return a value in the callback's response argument
+   * @function getId
+   * @param {SubmoduleParams} [configParams]
```
Comment (smenzer) [inline modules/fabrickIdSystem.js:38]:
```suggestion
   * @param {SubmoduleConfig} [config]
```

File: modules/fabrickIdSystem.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+   * @param {ConsentData}
+   * @param {Object} cacheIdObj - existing id, if any consentData]
+   * @returns {IdResponse|undefined}
+   */
+  getId(configParams, consentData, cacheIdObj) {
```
Comment (smenzer) [inline modules/fabrickIdSystem.js:43]:
```suggestion
  getId(config, consentData, cacheIdObj) {
    const configParams = (config && config.params) || {};
```

**Review by smenzer — COMMENTED** [review]
please note that after https://github.com/prebid/Prebid.js/pull/5803, the first config sent into `getId()`, `decode()`, and `extendId()` now includes the entire config object, not just params, so please make sure to account for that in this PR so your code works properly.

**Comment by andersonbd1** [comment]
Thanks for the heads up on #5803 , @smenzer .  I have updated accordingly and also submitted the Docs PR here:
https://github.com/prebid/prebid.github.io/pull/2399

**Comment by smenzer** [comment]
hi @andersonbd1 - I don't see the updates in this PR, did you push your changes? Also make sure to update your tests, I made some suggestions for the code, but I didn't go through all the tests

**Comment by andersonbd1** [comment]
@smenzer - you are correct.  I pushed my code to the wrong branch.  I made the same change you suggested (and from looking at the changes to other modules).  The tests should be unaffected.

**Comment by andersonbd1** [comment]
ok - tests are now passing.

**Review by smenzer — APPROVED** [review]
LGTM

**Pull request merged**